backend/apps/dx_testing/views.py

```
@@ -0,0 +1,116 @@
+            elif step.get('duration', 0) > 5000:  # >5 seconds is slow
+                friction_points.append({
+                    'step': step.get('name', ''),
+                    'warning': f"Slow step: {step.get('duration')}ms",
+                })
+        
+        # Generate recommendations
+        recommendations = self._generate_recommendations(friction_points)
+        
+        # Create test run
+        test_run = DXTestRun.objects.create(
+            run_id=data.get('runId', f"dx_{time.time()}"),
```

Ruff (0.15.21)

[error] Undefined name `time`

(F821)